### ISSUE TITLE
mip level / array layer naming is inconsistent

### ISSUE THREAD

**Issue by Kangz** [gpuweb/gpuweb#158]
Feedback we got internally is that the naming for mip levels and array layers is inconsistent and a bit confusing:

```
partial dictionary WebGPUTextureDescriptor {
    u32 arraySize;
    u32 levelCount;
};
partial dictionary WebGPUTextureCopyView {
    u32 level;
    u32 slice;
};
partial dictionary WebGPUTextureViewDescriptor {
    u32 baseMipLevel;
    u32 levelCount;
    u32 baseArrayLayer;
    u32 layerCount;
};
```

The suggestion was to rename the like this:

```
partial dictionary WebGPUTextureDescriptor {
    u32 arrayLayerCount;
    u32 mipLevelCount;
};
partial dictionary WebGPUTextureCopyView {
    u32 mipLevel;
    u32 arrayLayer;
};
partial dictionary WebGPUTextureViewDescriptor {
    u32 baseMipLevel;
    u32 mipLevelCount;
    u32 baseArrayLayer;
    u32 arrayLayerCount;
};
```

Which I think is great. Basically it makes it so we always talk about " mipLevel" and "arrayLayer" and never just "level" and "layer". It also makes "slice" consistent.

WDYT?

**Comment by kainino0x** [comment]
+1. btw, this naming seems to match Vulkan exactly.

**Comment by magcius** [comment]
How would one express a single face in a cubemap array in this scheme?

**Comment by kainino0x** [comment]
AFAICT all this suggestion does is rename fields. If there's a problem with expressing cubemap faces then can you open another issue?

**Comment by Kangz** [comment]
Fixed by #175
